lib/workers/repository/update/branch/auto-replace.ts

```diff
@@ -219,13 +219,7 @@ export async function doAutoReplace(
         );
       }
     }
-    if (!firstUpdate && (await confirmIfDepUpdated(upgrade, existingContent))) {
```

Comment:
This conditional check was causing a lot of heartache with this change.  the confirmIfDepUpdated check is run 3 other times during the process, and i couldn't see what value it was adding here.  Keen on feedback from the maintainers on this change.  

I did spend a fair amount of time trying to leave this check in-place and tweak the logic inside confirmIfDepUpdated, but that uncovered even more side-effects

Reply:
i think this is for cases where multiple deps are using a shared variable. 🤔

Reply:
Will await https://github.com/renovatebot/renovate/pull/20627/files being resolved - since i now think this guard can be added back after the `confirmIfDepUpdated` refactor